### PR TITLE
chore: remove new_friend_requests feature flag

### LINKED ISSUE #4124: Remove new friend requests & new passport feature flags
In order to reduce the effort of mantaining legacy flows, remove the code of the old friend requests & old passports, keeping the new flows implementation.

Acceptance criteria:

- [ ] remove the code of the legacy friend requests
- [ ] remove the code of the legacy passport

### PR DESCRIPTION
**Pull request by lorux0**
## What does this PR change?

`Fixes #4124 `
Removes deprecated code from the old friend request flow.

## How to test the changes?

1. Launch the explorer
2. Check that all friendship interactions are working as expected (adding, cancelling, removing, rejecting)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 19158ed</samp>

This pull request removes the obsolete feature flag `new_friend_requests` and refactors the code to use the new logic and controllers for handling friend requests in the friends HUD, the chat notification HUD, the passport HUD, the player info card HUD, and the user context menu. It also deletes the unused `WebInterfaceFriendsApiBridgeProxy` class and its meta file, and simplifies the plugin system factory and the service locator factory.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Factories/ServiceLocatorFactory/ServiceLocatorFactory.cs [hunk 1]
```diff
@@ -112,13 +112,12 @@ public static ServiceLocator CreateDefault()
 
             result.Register<IFriendsController>(() =>
             {
-                // TODO (NEW FRIEND REQUESTS): remove when the kernel bridge is production ready
+                // TODO: remove when the all the friendship responsibilities are migrated to unity
```
Comment (aixaCode) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Factories/ServiceLocatorFactory/ServiceLocatorFactory.cs:115]:
```suggestion
                // TODO: remove when all the friendship responsibilities are migrated to unity
```
What still is missing?

**Review by aixaCode — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Factories/ServiceLocatorFactory/ServiceLocatorFactory.cs (hunk 1 above)
Comment (lorux0) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Factories/ServiceLocatorFactory/ServiceLocatorFactory.cs:115]: We still use `use-social-client` which fallbacks to the WebInterface flow, for example for retrieval of friends with direct messages

**Review by Kinerius — APPROVED** [review]
So nice to see so much code being deleted :D

**Review by anicalbano — APPROVED** [review]
✅ Tested and approved by QA

**Pull request merged**